### PR TITLE
Patch absl after download to work around an nvcc bug.

### PR DESCRIPTION
**Pull request by bas-aarts**
The nvcc bug will be fixed in the CUDA 10.2 release.

### CODE REVIEW

File: tensorflow/workspace.bzl [hunk 1]
```diff
@@ -157,8 +157,9 @@ def tf_repositories(path_prefix = "", tf_repo_name = ""):
     tf_http_archive(
         name = "com_google_absl",
         build_file = clean_dep("//third_party:com_google_absl.BUILD"),
-        # TODO: Remove the patch when https://github.com/abseil/abseil-cpp/issues/326 is resolved.
-        patch_file = clean_dep("//third_party:com_google_absl_fix_mac_build.patch"),
+        # TODO: Remove the patch when https://github.com/abseil/abseil-cpp/issues/326 is resolved
+        # and when TensorFlow buils against CUDA 10.2
```
Comment (r4nt) [inline tensorflow/workspace.bzl]: s/buils/builds/

File: third_party/com_google_absl_fix_mac_and_nvcc_build.patch [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+     visibility = ["//visibility:public"],
+     deps = [":civil_time"],
+ )
+--- ./absl/strings/string_view.h	2019-09-23 13:20:52.000000000 -0700
++++ ./absl/strings/string_view.h.fixed	2019-09-23 13:20:48.000000000 -0700
+@@ -492,7 +492,14 @@
+       (std::numeric_limits<difference_type>::max)();
+ 
+   static constexpr size_type CheckLengthInternal(size_type len) {
++#if defined(__NVCC__) && !defined(NDEBUG)
```
Comment (r4nt) [inline third_party/com_google_absl_fix_mac_and_nvcc_build.patch]: Can we add a condition on < cuda 10.2 in the #if so we actually test for regressions once we compile with 10.2?

**Review by r4nt — CHANGES_REQUESTED** [review]

File: tensorflow/workspace.bzl (hunk 1 above)
Comment (bas-aarts) [reply tensorflow/workspace.bzl]: done

File: third_party/com_google_absl_fix_mac_and_nvcc_build.patch (hunk 2 above)
Comment (bas-aarts) [reply third_party/com_google_absl_fix_mac_and_nvcc_build.patch]: done

**Review by r4nt — DISMISSED** [review]

**Comment by gbaned** [comment]
@bas-aarts Can you please resolve conflicts? Thanks!

**Comment by bas-aarts** [comment]
@mihaimaruseac , @r4nt, in the hope to avoid another merge conflict for this PR, any change to have someone review this? Thanks.

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**